src/extensions/replay/sessionrecording.ts

```diff
@@ -406,7 +407,7 @@ export class SessionRecording {
                     reason: 'user activity',
                     type: event.type,
                 })
-                this._tryTakeFullSnapshot()
+                returningFromIdle = true
```

Comment:
as I added tracking of full snapshots in the tests it became clear there were routes through the code where we might take two full snapshots

(despite worrying about taking none full snapshots I tidied this up at the same time)